### PR TITLE
Add `arrayIncludes()`

### PR DESCRIPTION
**Pull request by ifiokjr**
### Description

This adds an improved `arrayIncludes` function which acts as a type guard.

```ts
import {arrayIncludes} from 'ts-extras';

const values = ['a', 'b', 'c'] as const;
declare const valueToCheck: unknown;
let value: typeof values[number];

if (arrayIncludes(values, valueToCheck)) {
  // Now we know that the value is of type `typeof values[number]`.
  value = valueToCheck
} else {
  // @ts-expect-error
  value = valueToCheck
}
```

### CODE REVIEW

File: source/array-includes.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/**
+A better version `Array.prototype.includes` which allows other types to be included and also acts as a type guard when the provided array is of a specific type.
+
+This is needed because `Array.prototype.includes` doesn't allow for other types to be included.
```
Comment (sindresorhus) [inline source/array-includes.ts:4]: Can you link to a TS issue about this?

File: source/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
+export {arrayIncludes} from './array-includes.js';
+export {asMutable} from './as-mutable.js';
+export {assertError} from './assert-error.js';
 export {isDefined} from './is-defined.js';
 export {isEmpty} from './is-empty.js';
-export {assertError} from './assert-error.js';
-export {asMutable} from './as-mutable.js';
```
Comment (sindresorhus) [inline source/index.ts:4]: Don't make unrelated changes.

File: test/array-includes.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	const invalidValue: unknown = 'd';
+	let testValueType: typeof values[number];
+
+	t.is(arrayIncludes(values, validValue), true);
+	t.is(arrayIncludes(values, invalidValue), false);
+
+	if (arrayIncludes(values, validValue)) {
+		testValueType = validValue;
+	} else {
+		doNothing(); // Removes the `unicorn/prefer-ternary` failure;
```
Comment (sindresorhus) [inline test/array-includes.ts:16]: Better to use an ESLint ignore comment.

File: test/array-includes.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import test from 'ava';
+import {arrayIncludes} from '../source/index.js';
+
+test('arrayIncludes()', t => {
+	const values = ['a', 'b', 'c'] as const;
+	const validValue: unknown = 'a';
+	const invalidValue: unknown = 'd';
+	let testValueType: typeof values[number];
+
+	t.is(arrayIncludes(values, validValue), true);
```
Comment (sindresorhus) [inline test/array-includes.ts:10]: Use `t.true`

File: test/array-includes.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import test from 'ava';
+import {arrayIncludes} from '../source/index.js';
+
+test('arrayIncludes()', t => {
+	const values = ['a', 'b', 'c'] as const;
```
Comment (sindresorhus) [inline test/array-includes.ts:5]: Can you also do an assertions that confirms that it preserves the readonly-ness of an array?

File: source/array-includes.ts (hunk 1 above)
Comment (ifiokjr) [reply source/array-includes.ts:4]: The proposal was rejected in this issue: https://github.com/microsoft/TypeScript/issues/26255#issuecomment-748211891

File: test/array-includes.ts (hunk 5 above)
Comment (ifiokjr) [reply test/array-includes.ts:5]: I'm not sure how this would affect the `readonly` status of the array. The type-guard only applies to the second argument. 

Would you want something like this?

```ts
// @ts-expect-error
values.push
```

Comment (sindresorhus) [reply test/array-includes.ts:5]: I just commented on an arbitrary line. The content above is not related to my comment.

Comment (sindresorhus) [reply test/array-includes.ts:5]: > Would you want something like this?

Yes

Comment (ifiokjr) [reply test/array-includes.ts:5]: I added something which, I hope, is not too ugly.
